pyfar/constants/frequency_weighting.py

```diff
@@ -0,0 +1,198 @@
+"""File containing all frequency weighting functions."""
+from typing import Literal, Union
+import numpy as np
+
+# Constants for the weighting curve formulas.
+# See appendix E in IEC 61672-1.
+_F_1 = 20.598997057618316
+_F_2 = 107.65264864304629
+_F_3 = 737.8622307362901
+_F_4 = 12194.217147998012
+_A_1000 = -2
+_C_1000 = -0.062
+
+# Constants for nominal band corrections, taken from table 3 in IEC 61672-1.
+# Could be replaced/combined with a dedicated frequency_bands_nominal()
+# function later.
+_NOMINAL_THIRDBAND_FREQUENCIES = np.array([
+    10, 12.5, 16, 20, 25, 31.5, 40, 50, 63, 80, 100, 125, 160, 200, 250, 315,
+    400, 500, 630, 800, 1000, 1250, 1600, 2000, 2500, 3150, 4000, 5000, 6300,
+    8000, 10000, 12500, 16000, 20000,
+])
+_THIRDBAND_WEIGHTINGS_A = np.array([
+    -70.4, -63.4, -56.7, -50.5, -44.7, -39.4, -34.6, -30.2, -26.2, -22.5,
+    -19.1, -16.1, -13.4, -10.9, -8.6, -6.6, -4.8, -3.2, -1.9, -0.8,
+    0.0, 0.6, 1.0, 1.2, 1.3, 1.2, 1.0, 0.5, -0.1, -1.1,
+    -2.5, -4.3, -6.6, -9.3,
+])
+_THIRDBAND_WEIGHTINGS_C = np.array([
+    -14.3, -11.2, -8.5, -6.2, -4.4, -3.0, -2.0, -1.3, -0.8, -0.5,
+    -0.3, -0.2, -0.1, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+    0.0, 0.0, -0.1, -0.2, -0.3, -0.5, -0.8, -1.3, -2.0, -3.0, -4.4,
+    -6.2, -8.5, -11.2,
+])
+
+
+def _calculate_A_weighted_level(f: Union[float, np.ndarray],
+                                ) -> Union[float, np.ndarray]:
+    """
+    Calculates the level correction in dB of a frequency component
+    when using the A weighting according to IEC 61672-1.
+    """
+    # formula (E.6) in the standard
+    bracket_term = (_F_4**2 * f**4) / ((f**2 + _F_1**2) * (f**2 + _F_2**2)
+                                      ** 0.5 * (f**2 + _F_3**2)**0.5
+                                      * (f**2 + _F_4**2))
+    return 10 * np.log10(bracket_term**2) - _A_1000
+
+
+def _calculate_C_weighted_level(f: Union[float, np.ndarray],
+                                ) -> Union[float, np.ndarray]:
+    """
+    Calculates the level correction in dB of a frequency component
+    when using the C weighting according to IEC 61672-1.
+    """
+    # formula (E.1) in the standard
+    bracket_term = (_F_4**2 * f**2) / ((f**2 + _F_1**2) * (f**2 + _F_4**2))
+    return 10 * np.log10(bracket_term**2) - _C_1000
+
+
+def frequency_weighting_curve(weighting: Literal["A", "C"],
+                              frequencies: list[float]):
+    """
+    Calculates the level correction in dB of a frequency component when using
+    the A or C weighting defined in IEC 61672-1.
+
+    Parameters
+    ----------
+    weighting: str ('A' or 'C')
+        Which frequency weighting to use.
+
+    frequencies: array-like
+        The frequencies at which to evaluate the weighting curve.
+
+    Returns
+    -------
+    weights: NDArray
```

Comment:
```suggestion
    weights: numpy.ndarray[float]
```
I think something like this should render a link to the respecive numpy docs.
Again, there are a couple of occurences, where this could be changed below.